packages/exchange-connector/test/exchangeDownloader.test.ts

```diff
@@ -67,84 +68,76 @@ describe("Test downloadRestApi", () => {
     const api = _.cloneDeep(REST_API);
 
     return downloadRestApi(api, tmpDir.name).then((res: Response) => {
-      expect(res.status).to.be.equal(200);
+      expect(res.status).to.equal(200);
     });
   });
 
-  it("can download single file, no download dir specified", async () => {
+  it("can download a single file even when no download dir is specified", async () => {
     nock("https://somewhere")
       .get("/fatraml.zip")
       .reply(200);
 
     const api = _.cloneDeep(REST_API);
 
     return downloadRestApi(api).then((res: Response) => {
-      expect(res.status).to.be.equal(200);
+      expect(res.status).to.equal(200);
     });
   });
 
-  it("throws if no fat raml", async () => {
-    const tmpDir = tmp.dirSync();
-
-    nock("https://somewhere")
-      .get("/fatraml.zip")
-      .reply(200);
-
+  it("doesn't return anything if fat raml information is missing", async () => {
     const api = _.cloneDeep(REST_API);
+    delete api.fatRaml.createdDate;
 
-    delete api.fatRaml;
-
-    return expect(downloadRestApi(api, tmpDir.name)).to.be.rejectedWith(
-      "Fat RAML download information for test-api is missing"
-    );
+    return expect(downloadRestApi(api)).to.eventually.be.undefined;
   });
 });
 
-describe("Test downloadRestApis", () => {
-  afterEach(nock.cleanAll);
+describe("downloadRestApis", () => {
+  beforeEach(sinon.restore);
```

Comment:
Why did you switch from nock to sinon here?

Reply:
To stub the helper functions, sinon was only mocking the calls made by the helper functions. I believe the tests shouldn't be invoking the helper functions unless they are the ones under test.